ibis/backends/druid/compiler.py

```diff
@@ -1,33 +1,232 @@
 from __future__ import annotations
 
-import contextlib
+from functools import singledispatchmethod
 
-import sqlalchemy as sa
+import sqlglot as sg
+import sqlglot.expressions as sge
+import toolz
+from sqlglot import exp
+from sqlglot.dialects import Postgres
+from sqlglot.dialects.dialect import rename_func
 
-import ibis.backends.druid.datatypes as ddt
-from ibis.backends.base.sql.alchemy import AlchemyCompiler, AlchemyExprTranslator
-from ibis.backends.druid.registry import operation_registry
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+from ibis.backends.base.sqlglot.compiler import NULL, SQLGlotCompiler
+from ibis.backends.base.sqlglot.datatypes import DruidType
 from ibis.expr.rewrites import rewrite_sample
 
 
-class DruidExprTranslator(AlchemyExprTranslator):
-    _registry = operation_registry.copy()
-    _rewrites = AlchemyExprTranslator._rewrites.copy()
-    _dialect_name = "druid"
+# Is postgres the best dialect to inherit from?
```

Comment:
Do you want to keep this comment? :)

Reply:
I mean, it's still an open question

Reply:
Apparently it is working, so I'd remove the comment.